src/App.tsx

```diff
@@ -1,41 +1,103 @@
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
-
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
-
-      <div className="App__content">
-        <PostDetails />
-      </div>
-    </main>
-  </div>
-);
+const App: React.FC = () => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState('0');
+  const [selectedPostId, setSelectedPostId] = useState(0);
+
+  useEffect(() => {
+    const fetchUsers = async () => {
+      try {
+        const usersFromServer = await getAllUsers();
+
+        setUsers(usersFromServer);
+      } catch {
+        Promise.reject(new Error('error'));
+      }
+    };
+
+    fetchUsers();
+  }, []);
+
+  useEffect(() => {
+    const fetchPosts = async () => {
+      const postsFromServer = await getAllPosts();
+
+      setPosts(postsFromServer);
+    };
+
+    const fetchUserPosts = async () => {
+      const userPosts = await getUserPosts(selectedUserId);
+
+      setPosts(userPosts);
+    };
+
+    try {
+      (selectedUserId === '0' ? fetchPosts : fetchUserPosts)();
+    } catch {
+      Promise.reject(new Error('error'));
+    }
+  }, [selectedUserId]);
+
+  return (
+    <div className="App">
+      <header className="App__header">
+        <label htmlFor="userSelector">
+          Select a user: &nbsp;
+
+          <select
+            className="App__user-selector"
+            id="userSelector"
+            value={selectedUserId}
+            onChange={(event) => {
+              setSelectedUserId(event.target.value);
+              setSelectedPostId(0);
+            }}
+          >
+            <option value="0">All users</option>
+            {users.map(user => (
+              <option
+                value={user.id}
+                key={user.id}
+              >
+                {user.name}
+              </option>
+            ))}
+          </select>
+        </label>
+      </header>
+
+      <main className="App__main">
+        <div className="App__sidebar">
+          <PostsList
+            posts={posts}
+            selectedPostId={selectedPostId}
+            setSelectedPostId={setSelectedPostId}
+          />
+        </div>
+
+        <div className="App__content">
+          {selectedPostId ? (
+            <PostDetails
+              post={posts.filter(post => post.id === selectedPostId)[0]}
+              selectedPostId={selectedPostId}
+            />
```

Comment:
according to the task you need to fetch post details separately by it's post id, not just filter all post and pass it as prop
> Create getPostDetails(postId) method in /src/api/posts.ts and load post details from /posts/:postId